Applications/Spire/Source/KeyBindings/TaskKeyBindingsTableModel.cpp

```diff
@@ -0,0 +1,344 @@
+      index.row() < static_cast<int>(m_key_bindings.size())) {
+    switch(static_cast<Columns>(index.column())) {
+      case Columns::NAME:
+        return QString::fromStdString(
+          m_key_bindings[index.row()].m_action.m_name);
+      case Columns::SECURITY:
+        {
+          auto& securities =
+            m_key_bindings[index.row()].m_region.GetSecurities();
+          if(!securities.empty()) {
+            return to_variant(*(securities.begin()));
+          }
+        }
+        break;
+      case Columns::DESTINATION:
+        return to_variant(m_key_bindings[index.row()].m_region);
+      case Columns::ORDER_TYPE:
+        if(auto type = m_key_bindings[index.row()].m_action.m_type) {
+          return to_variant(*type);
+        }
+        break;
+      case Columns::SIDE:
+        if(auto side = m_key_bindings[index.row()].m_action.m_side) {
+          return to_variant(*side);
+        }
+        break;
+      case Columns::QUANTITY:
+        if(auto quantity = m_key_bindings[index.row()].m_action.m_quantity) {
+          return to_variant(*quantity);
+        }
+        break;
+      case Columns::TIME_IN_FORCE:
+        {
+          auto time_in_force =
+            m_key_bindings[index.row()].m_action.m_time_in_force;
+          if(time_in_force &&
+              time_in_force->GetType() != TimeInForce::Type::NONE) {
+            return to_variant(*time_in_force);
+          }
+        }
+        break;
+      case Columns::CUSTOM_TAGS:
+        return "";
+      case Columns::KEY_BINDING:
+        return m_key_bindings[index.row()].m_sequence;
+      default:
+        return QVariant();
+    }
+  }
+  return QVariant();
+}
+
+Qt::ItemFlags TaskKeyBindingsTableModel::flags(
+    const QModelIndex& index) const {
+  if(static_cast<Columns>(index.column()) == Columns::CUSTOM_TAGS) {
+    return QAbstractItemModel::flags(index);
+  }
+  return QAbstractItemModel::flags(index) |= Qt::ItemIsEditable;
+}
+
+QVariant TaskKeyBindingsTableModel::headerData(int section,
+    Qt::Orientation orientation, int role) const {
+  if(orientation == Qt::Horizontal && role == Qt::DisplayRole) {
+    switch(static_cast<Columns>(section)) {
+      case Columns::NAME:
+        return tr("Name");
+      case Columns::SECURITY:
+        return tr("Security");
+      case Columns::DESTINATION:
+        return tr("Destination");
+      case Columns::ORDER_TYPE:
+        return tr("Order Type");
+      case Columns::SIDE:
+        return tr("Side");
+      case Columns::QUANTITY:
+        return tr("Quantity");
+      case Columns::TIME_IN_FORCE:
+        return tr("Time in Force");
+      case Columns::CUSTOM_TAGS:
+        return tr("Custom Tag");
+      case Columns::KEY_BINDING:
+        return tr("Key");
+      default:
+        return QVariant();
+    }
+  }
+  return QVariant();
+}
+
+bool TaskKeyBindingsTableModel::removeRows(int row, int count,
+    const QModelIndex &parent) {
+  if(row >= static_cast<int>(m_key_bindings.size())) {
```

Comment:
As a general principle, prefer widening casts to narrowing ones. The latter are inherently more dangerous.
For example, the proper way to do this check is
`row < 0 || static_cast<std::size_t>(row)  >= m_key_bindings.size()`
It does not make a difference as the number of bindings is highly unlikely to exceed the cardinality of `int`, but then it is a matter of coding style.